requests/__init__.py

```diff
@@ -48,11 +48,20 @@
 __copyright__ = 'Copyright 2016 Kenneth Reitz'
 
 # Attempt to enable urllib3's SNI support, if possible
-try:
-    from .packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
-except ImportError:
-    pass
+# If the user has a recent version of OpenSSL, this
+# isn't needed as Python's core ssl module handles SNI, etc correctly.
+# If the requirements of urllib3.contrib.pyopenssl change,
+# this version should be modified appropraitely.
+#
+# See https://github.com/kennethreitz/requests/issues/3213 as well.
+import ssl
+
+if ssl.OPENSSL_VERSION_INFO < (1, 0, 1):
```

Comment:
Sorry - let's unwind the stack a bit here. I have an environment where pyOpenSSL is present, but I don't want to use this automatically imported module, because I'm guaranteed to have a modern Python and a recent version of OpenSSL. I want to reduce imports and import time. Loading `contrib.pyopenssl` brings in a bunch of code I don't want, including `pkg_resources` via `cryptography`.  The OpenSSL version check was a potential work-around proposed by @sigmavirus24. I just want a mechanism to opt-out of trying to do this import at all.